recipe/recipe.yaml

```diff
@@ -0,0 +1,63 @@
+context:
+  name: libedit
+  version: 3.1
+  date: 20240808
+  sha256: 5f0573349d77c4a48967191cdd6634dd7aa5f6398c6a57fe037cc02696d6099f
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}.${{ date }}
+
+source:
+  url: http://thrysoee.dk/editline/${{ name }}-${{ date }}-${{ version }}.tar.gz
+  sha256: ${{ sha256 }}
+  # file_name: ${{ name }}-${{ date }}-${{ version }}.tar.gz
+
+build:
+  number: 0
+  skip: win
+
+requirements:
+  build:
+    - if: unix
+      then: libtool
+    - ${{ compiler('c') }}
+    - ${{ stdlib("c") }}
+    - automake
+    - perl
+    - make
+    - m4
+  host:
+    - ncurses
+  run:
+    - ncurses
+  run_exports:
+    - ${{ pin_subpackage('libedit', upper_bound='x.x') }}
+
+tests:
+  - script:
+      - if: osx
+        then: test -f $PREFIX/lib/libedit.dylib
```

Comment:
you could also use `${SHLIB_EXT}` (ie. `$PREFIX/lib/libedit${SHLIB_EXT}`). Then it works on both unixes.